### PR TITLE
Add Handwired/Curiosity [Keyboard Addition]

### PR DESCRIPTION
**Pull request by Spaceman**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/curiosity/curiosity.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+#define SMALL_MODS( \
```
Comment (drashna) [inline keyboards/handwired/curiosity/curiosity.h:20]:
```suggestion
#define LAYOUT_small_mods( \
```

File: keyboards/handwired/curiosity/curiosity.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b,  k3a, \
+      k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, k39, \
+       k30,  k31,  k32,     k33,       k34,    k35, k36, k37, k38 \
+) { \
+    { k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b }, \
+    { k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b }, \
+    { k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b }, \
+    { k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b }, \
+}
+#define LARGE_MODS( \
```
Comment (drashna) [inline keyboards/handwired/curiosity/curiosity.h:31]:
```suggestion
#define LAYOUT_large_mods( \
```

File: keyboards/handwired/curiosity/info.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "keyboard_name": "Curiosity",
+  "url": "",
+  "maintainer": "Spaceman",
+  "width": 13.5,
+  "height": 4,
+  "layouts": {
+    "SMALL_MODS": {
```
Comment (drashna) [inline keyboards/handwired/curiosity/info.json:8]:
```suggestion
    "LAYOUT_small_mods": {
```

File: keyboards/handwired/curiosity/info.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      "layout": [
+        {"x":0, "y":0}, {"x":1, "y":0}, {"x":2, "y":0}, {"x":3, "y":0}, {"x":4, "y":0}, {"x":5, "y":0}, {"x":6, "y":0}, {"x":7, "y":0}, {"x":8, "y":0}, {"x":9, "y":0}, {"x":10, "y":0}, {"x":11, "y":0}, {"x":12, "y":0, "w":1.5},
+        {"x":0, "y":1, "w":1.25}, {"x":1.25, "y":1}, {"x":2.25, "y":1}, {"x":3.25, "y":1}, {"x":4.25, "y":1}, {"x":5.25, "y":1}, {"x":6.25, "y":1}, {"x":7.25, "y":1}, {"x":8.25, "y":1}, {"x":9.25, "y":1}, {"x":10.25, "y":1}, {"x":11.25, "y":1}, {"x":12.25, "y":1, "w":1.25},
+        {"x":0, "y":2, "w":1.5}, {"x":1.5, "y":2}, {"x":2.5, "y":2}, {"x":3.5, "y":2}, {"x":4.5, "y":2}, {"x":5.5, "y":2}, {"x":6.5, "y":2}, {"x":7.5, "y":2}, {"x":8.5, "y":2}, {"x":9.5, "y":2}, {"x":10.5, "y":2}, {"x":11.5, "y":2}, {"x":12.5, "y":2},
+        {"x":0.75, "y":3, "w":1.25}, {"x":2, "y":3, "w":1.25}, {"x":3.25, "y":3, "w":1.25}, {"x":4.5, "y":3, "w":2}, {"x":6.5, "y":3, "w":2.25}, {"x":8.75, "y":3}, {"x":9.75, "y":3}, {"x":10.75, "y":3}, {"x":11.75, "y":3}
+      ]
+    },       
+    "LARGE_MODS": {
```
Comment (drashna) [inline keyboards/handwired/curiosity/info.json:16]:
```suggestion
    "LAYOUT_large_mods": {
```

File: keyboards/handwired/curiosity/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = SMALL_MODS(
```
Comment (drashna) [inline keyboards/handwired/curiosity/keymaps/default/keymap.c:20]:
```suggestion
    [0] = LAYOUT_small_mods(
```

File: keyboards/handwired/curiosity/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_DEL, KC_BSPC,
+        KC_TAB, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_ENT,
+        KC_NO, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, MO(1), KC_RSFT,
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_SPC, KC_LEFT, KC_DOWN, KC_UP, KC_RGHT
+    ),
+
+	[1] = SMALL_MODS(
```
Comment (drashna) [inline keyboards/handwired/curiosity/keymaps/default/keymap.c:27]:
```suggestion
    [1] = LAYOUT_small_mods(
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/curiosity/rules.mk [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/handwired/curiosity/rules.mk:11]:
```suggestion
```

File: keyboards/handwired/curiosity/rules.mk [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+LTO_ENABLE = no             # Use link time optimization
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```
Comment (fauxpark) [inline keyboards/handwired/curiosity/rules.mk:33]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = no             # Use link time optimization
```

File: keyboards/handwired/curiosity/rules.mk [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
```
Comment (fauxpark) [inline keyboards/handwired/curiosity/rules.mk:26]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/curiosity/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,33 @@
+/* Copyright 2019 marhalloweenvt
```
Comment (noroadsleft) [inline keyboards/handwired/curiosity/keymaps/default/keymap.c:1]: marhalloweenvt?

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/handwired/curiosity/keymaps/default/keymap.c (hunk 10 above)
Comment (Spaceman) [reply keyboards/handwired/curiosity/keymaps/default/keymap.c:1]: Whoops my bad just copied from another keyboard

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**